src/libs/ReportUtils.ts

```diff
@@ -2296,6 +2296,7 @@ function getReportPreviewMessage(
     isPreviewMessageForParentChatReport = false,
     policy: OnyxEntry<Policy> = null,
     isForListPreview = false,
+    originalReportAction: OnyxEntry<ReportAction> | EmptyObject = reportAction,
```

Comment:
You are correct that they are very similar but there is one case where the values differ.

In general, `reportAction` **should** be the message for which the preview is being generated. However, a different action is passed for `reportAction` when generating the LHN message for a report preview (a message grouping multiple money requests).

https://github.com/Expensify/App/blob/49cb3b1502856a42f9f4ff5c6afbe7b19712153c/src/libs/OptionsListUtils.ts#L558-L573

I believe this is a bug and should be fixed separately.

To generate the correct preview text for a report preview action, we need a way to access the original preview action so that we can determine the number of child reports in the preview. To solve this, I pass the original as `originalReportAction` while leaving `reportAction` unchanged. (Fixing the value of `reportAction` would blow out the scope of this PR, particularly around QA/validation, so I've decided not to do that.)

**Visual Example**

Here is an example of a "report preview" message. In this case, `reportAction` **should** be the last message. However, in practice, `reportAction` is one of the child money request reports. To correct for this, the actual preview message is passed as `originalReportAction`.

<img width="1042" alt="image" src="https://github.com/Expensify/App/assets/1534986/eb0c9d4f-4603-4332-b196-18aa9cfdf2d5">

Reply:
Okay, done with one tweak. I renamed `reportAction ` to `iouReportAction` as I think this will be less confusing to readers. Hopefully the comments clarify why this name makes sense.

Reply:
The previous comment suggested `reportAction` could be either a report preview or an IOU report action so perhaps that comment is incorrect.

> `@param [reportAction] This can be either a report preview action or the IOU action`

I did not want to rename `reportAction` to `previewReportAction` because I that name implies the action is a preview (ie, a report preview). This would be confusing because the value is not a report preview action.

I chose `iouReportAction` because it seemed the report action could only be an IOU Report Action. However, if that's not true and it can also be a normal DM, then I no longer believe `iouReportAction` is the better name.

Do you still consider `previewReportAction` to be the correct choice?